applications/FluidDynamicsApplication/python_scripts/fluid_solver.py

```diff
@@ -223,6 +212,27 @@ def _ExecuteCheckAndPrepare(self):
 
         check_and_prepare_model_process_fluid.CheckAndPrepareModelProcess(self.main_model_part, prepare_model_part_settings).Execute()
 
+    def _SetAndFillBuffer(self):
+        # Set the buffer size for the nodal solution steps data.
+        # Existing nodal solution step data may be lost.
+        required_buffer_size = self.GetMinimumBufferSize()
+        current_buffer_size = self.main_model_part.GetBufferSize()
+        buffer_size = max(current_buffer_size, required_buffer_size)
+        self.main_model_part.SetBufferSize(buffer_size)
+
+        # Cycle the buffer. This sets all historical nodal solution step data to
+        # the current value and initializes the time stepping in the process info.
+        step = -buffer_size
+        delta_time = self._ComputeInitialDeltaTime()
+        init_time = self.main_model_part.ProcessInfo[KratosMultiphysics.TIME]
```

Comment:
I think the initial time is not yet set and will return always 0. Convergence problems may appear if the initial time is different than 0

Reply:
if the implementation of @rubenzorrilla is done exactly in the same way as n the structrual app then it works fine.
The solution there has existed since forever and has always worked afaik

Reply:
have you checked with a different initial time? (especially with explicit strategies)

Reply:
In restarts the initial time is different from 0
Also the same is used for the explicit structural simulations

or what exactly do you mean?

Reply:
if it works with a non restarted simulation with initial time different from 0:
```json
    "problem_data"     : {
        "start_time"    : 100.0,
        "end_time"      : 100.1,
        ...
```

Reply:
yes this is what we always do for restarts

Reply:
actually I confused it, sorry my bad
In a restart this fct is not called at all, then the current time is gotten from the restart file
https://github.com/KratosMultiphysics/Kratos/blob/1a8bc639d77a054eb7f15c1f122d1ef7135e8fb4/kratos/python_scripts/analysis_stage.py#L104-L105

actually looking at your PR #9267 I am not sure what will happen with a `start_time` unequal to zero :/
in this case the comment from above is correct

Reply:
> In restarts the initial time is different from 0 Also the same is used for the explicit structural simulations
> 
> or what exactly do you mean?

I can confirm this is the current behavior.